server/task_scheduler.go

```diff
@@ -89,8 +89,8 @@ func (s *TaskScheduler) Run(ctx context.Context, wg *sync.WaitGroup) {
 						continue
 					}
 
-					if _, err := s.server.ScheduleNextTaskIfNeeded(ctx, pipeline); err != nil {
-						log.Error("Failed to schedule next running task",
+					if err := s.server.ScheduleActiveStageTask(ctx, pipeline); err != nil {
+						log.Error("Failed to schedule the tasks in the active stage",
```

Comment:
```suggestion
						log.Error("Failed to schedule tasks in the active stage",
```
nit: less verbose.

Reply:
Pardon my poor English, I think the "the" is necessary here, is it?